core/server/worker/src/main/java/alluxio/worker/block/BlockMasterClient.java

```diff
@@ -310,7 +310,9 @@ public void register(final long workerId, final List<String> storageTierAliases,
 
     final RegisterWorkerPOptions options =
         RegisterWorkerPOptions.newBuilder().addAllConfigs(configList)
-            .setBuildVersion(buildVersion).build();
+            .setBuildVersion(buildVersion)
+            .setNumVCpu(Runtime.getRuntime().availableProcessors())
```

Comment:
`Runtime.getRuntime().availableProcessors()` does this work in a container?
From https://github.com/Alluxio/alluxio/issues/17021 seems in some cases the issue still exists.

Reply:
yup i am well aware of this issue